Circulars: Enable advanced search

tylerbarna opened the pull request:
<!-- IMPORTANT!!! Aside from typographical corrections and minor changes, please consider creating an Issue before making a Pull Request. -->

# Description
This enables the advanced search functionality, which allows for lucene style searches, along with the associated documentation.

# Related Issue(s)
Resolves #2288 

# Testing
1. Navigate to Circulars archive
2. submit a lucene style query (example: `subject:GRB`)
3. Results should differ from current production results
4. Submit an invalid lucene style query (example: `subject:GRB OR`)
5. Results should still load but with a warning that it is an invalid search. Results should be similar to searching `subject:GRB or` on production since it is falling back on `multi_match`
6. Clicking links to the documentation in the Advanced Search dropdown or the search warning should direct to the subsection in the docs on advanced search.

tylerbarna committed:
remove CIRCULARS_LUCENE feature flag

tylerbarna committed:
refactor queryObj slightly

slightly shortens the queryObj definition while retaining existing function. the try statement could potentially be refactored somewhat, but a few variations I tried did not work as intended.

tylerbarna committed:
remove unused import

tylerbarna commented:
It might be good to hold off on merging this until announcing it as a feature

lpsinger commented:
> It might be good to hold off on merging this until announcing it as a feature (#2995)

We don't necessarily need to announce it first. Announcing will call attention to it, that's all.

Before we enable it in production, would you please team with someone else on Slack and try to enter a bunch of simple and advanced queries on dev and just check that there is not any glaringly incorrect behavior?

tylerbarna commented:
> > It might be good to hold off on merging this until announcing it as a feature (#2995)
> 
> We don't necessarily need to announce it first. Announcing will call attention to it, that's all.
> 
> Before we enable it in production, would you please team with someone else on Slack and try to enter a bunch of simple and advanced queries on dev and just check that there is not any glaringly incorrect behavior?

Sure, I have a meeting in a few minutes, but I'll check in to see who's free and try to get confirmation before the end of the week

tylerbarna committed:
remove documentation on wildcards pending further investigation

wildcard searches using * and ? operators, specifically in the use case of searching for somethinglike `subject:"GRB 23*", is not working as intended. However, wildcard searches are not necessary to enable the advanced search feature. Wildcard searches could be implemented in the future pending further investigation and depending on community need.

tylerbarna committed:
rerun checks

dakota002 commented:
> Before we enable it in production, would you please team with someone else on Slack and try to enter a bunch of simple and advanced queries on dev and just check that there is not any glaringly incorrect behavior?

I ran through the following queries Tyler provided me and got the numbers of results listed:
```
Swift: 16079 
subject:"Swift": 9561 
body:"Swift": 15958
subject:"Swift" AND body:"GRB": 8239
subject:"Swift" OR body:"GRB": 27609
subject:"Swift" AND (body:"GRB" OR body:"Gamma-ray burst"): 8254
subject:"LIGO" AND NOT body:"GRB": 2099
```

The results looked correct for what each query was asking too (correct matches based on subject/body and combinations of conditions )

tylerbarna committed:
update example submitter

tylerbarna committed:
update docs description

tylerbarna committed:
remove reference to lucene

tylerbarna committed:
run prettier
